community/kernel/src/main/java/org/neo4j/procedure/Procedure.java

```diff
@@ -101,4 +101,15 @@
 @Retention( RetentionPolicy.RUNTIME )
 public @interface Procedure
 {
+    /**
+     * The namespace and name for the procedure, as a period-separated
```

Comment:
Can we instead have separate elements `namespace` and `name`? This will make naming more flexible and reduce parsing in `ReflectiveProcedureCompiler#extractName(Class<?>, Method)`.